src/main/java/org/zendesk/client/v2/Zendesk.java

```diff
@@ -1431,48 +1416,34 @@ public static ObjectMapper createMapper() {
     private class PagedIterable<T> implements Iterable<T> {
 
         private final Uri url;
-        private final AsyncCompletionHandler<List<T>> handler;
-        private final int initialPage;
-        private int size = 0;
+        private final PagedAsyncCompletionHandler<List<T>> handler;
 
-        private PagedIterable(Uri url, AsyncCompletionHandler<List<T>> handler) {
-            this(url, handler, 1);
-        }
-
-        private PagedIterable(Uri url, AsyncCompletionHandler<List<T>> handler, int initialPage) {
+        private PagedIterable(Uri url, PagedAsyncCompletionHandler<List<T>> handler) {
             this.handler = handler;
             this.url = url;
-            this.initialPage = initialPage;
         }
 
         public Iterator<T> iterator() {
-            return new PagedIterator(initialPage);
+            return new PagedIterator(url);
         }
 
         private class PagedIterator implements Iterator<T> {
```

Comment:
Unfortunately PagedIterator is too tightly coupled to Zendesk and I didn't have time to refactor / extract it.